src/components/stepper/stepper/stepper.ts

```diff
@@ -0,0 +1,278 @@
+import type { SbbStepElement, SbbStepValidateEventDetails } from '../step/step';
+
+import style from './stepper.scss?lit&inline';
+
+const DEBOUNCE_TIME = 150;
+
+/**
+ * Provides a structured, step-by-step workflow for user interactions.
+ *
+ * @slot step-label - Use this slot to provide an `sbb-step-label`.
+ * @slot step - Use this slot to provide an `sbb-step`.
+ */
+@customElement('sbb-stepper')
+export class SbbStepperElement extends LitElement {
+  public static override styles: CSSResultGroup = style;
+
+  /**
+   * If set to true, only the current and previous labels can be clicked and selected.
+   */
+  @property({ type: Boolean }) public linear = false;
+
+  /**
+   * Overrides the behaviour of `orientation` property.
+   */
+  @property({ attribute: 'horizontal-from', reflect: true })
+  public get horizontalFrom(): SbbHorizontalFrom | undefined {
+    return this._horizontalFrom;
+  }
+  public set horizontalFrom(value: SbbHorizontalFrom) {
+    this._horizontalFrom = breakpoints.includes(value) ? value : undefined;
+    if (this._horizontalFrom) {
+      this._checkOrientation();
+    }
+  }
+  private _horizontalFrom?: SbbHorizontalFrom | undefined;
+
+  /**
+   * Steps orientation, either horizontal or vertical.
+   */
+  @property({ reflect: true })
+  public orientation: SbbOrientation = 'horizontal';
+
+  /**
+   * The currently selected step.
+   */
+  public get selected(): SbbStepElement | undefined {
+    return this.querySelector<SbbStepElement>('sbb-step[data-selected]') ?? undefined;
+  }
+
+  public set selected(step: SbbStepElement) {
+    this._select(step);
+  }
+
+  /**
+   * The currently selected step index.
+   */
+  public get selectedIndex(): number | undefined {
+    return this.selected ? this.steps.indexOf(this.selected) : undefined;
+  }
+
+  public set selectedIndex(index: number) {
+    this._select(this.steps[index]);
+  }
+
+  /**
+   * The steps of the stepper.
+   */
+  public get steps(): SbbStepElement[] {
+    return Array.from(this.querySelectorAll('sbb-step'));
+  }
+
+  /**
+   * Selects the next step.
+   */
+  public next(): void {
+    if (this.selectedIndex !== undefined) {
+      this._select(this.steps[this.selectedIndex + 1]);
+    }
+  }
+  /**
+   * Selects the previous step.
+   */
+  public previous(): void {
+    if (this.selectedIndex !== undefined) {
+      this._select(this.steps[this.selectedIndex - 1]);
+    }
+  }
+
+  /**
+   * Resets the form in which the stepper is nested or every form of each step, if any.
+   */
+  public reset(): void {
+    const closestForm = this.closest('form');
+    if (closestForm) {
+      closestForm.reset();
+    } else {
+      this.querySelectorAll('form').forEach((form) => form.reset());
+    }
+    this.selectedIndex = 0;
+  }
+
+  private _loaded: boolean = false;
+  private _abort = new ConnectedAbortController(this);
+  private _resizeObserverTimeout: ReturnType<typeof setTimeout> | null = null;
```

Comment:
I would move this before the `next` method but maybe we should discuss this in with team